tests/e2e/config_secret_utils.go

```diff
@@ -0,0 +1,676 @@
+	inSecureFlag bool, testUser string, password string, vCenterPort string,
+	dataCenter string, sshClientConfig *ssh.ClientConfig,
+	masterIp string) error {
+	conf := `tee csi-vsphere.conf >/dev/null <<EOF
+[Global]
+cluster-id = "%s"
+cluster-distribution = "%s"
+[VirtualCenter "%s"]
+insecure-flag = "%t"
+user = "%s"
+password = "%s"
+port = "%s"
+datacenters = "%s"`
+
+	createConf := fmt.Sprintf(conf, clusterName, clusterDistribution, vCenterIP, inSecureFlag, testUser, password,
+		vCenterPort, dataCenter)
+	framework.Logf("vsphere config secret file: %s ", createConf)
+	result, err := sshExec(sshClientConfig, masterIp, createConf)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			conf, masterIp, err)
+	}
+	applyConf := "kubectl create secret generic vsphere-config-secret --from-file=csi-vsphere.conf " +
+		"-n vmware-system-csi"
+	framework.Logf("Create vsphere config secret: %s ", applyConf)
+	result, err = sshExec(sshClientConfig, masterIp, applyConf)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			applyConf, masterIp, err)
+	}
+	return nil
+}
+
+// deleteCsiVsphereConfFile method is used to delete csi vsphere conf file
+func deleteCsiVsphereConfFile(sshClientConfig *ssh.ClientConfig, masterIp string) error {
+	deleteConf := "kubectl delete secret vsphere-config-secret --namespace=vmware-system-csi"
+	framework.Logf("Delete vsphere config secret: %s ", deleteConf)
+	result, err := sshExec(sshClientConfig, masterIp, deleteConf)
+	if err != nil && result.Code != 0 {
+		fssh.LogResult(result)
+		return fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+			deleteConf, masterIp, err)
+	}
+	return nil
+}
+
+// restartCSIDriver method restarts the csi driver
+func restartCSIDriver(ctx context.Context, sshClientConfig *ssh.ClientConfig, masterIp string,
+	client clientset.Interface, namespace string) error {
+	ignoreLabels := make(map[string]string)
+	list_of_pods, err := fpod.GetPodsInNamespace(client, csiSystemNamespace, ignoreLabels)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	num_csi_pods := len(list_of_pods)
+
+	// collecting csi logs before restarting csi driver
+	ginkgo.By("Collecting csi logs before restarting csi driver")
+	collectCSILogs(ctx, client, sshClientConfig, masterIp, namespace)
+
+	// Restart CSI daemonset
+	restartCmd := []string{"rollout", "restart", "daemonset/vsphere-csi-node", "--namespace=" + csiSystemNamespace}
+	framework.RunKubectlOrDie(csiSystemNamespace, restartCmd...)
+	framework.Logf("Restarted csi driver: %s ", restartCmd)
+
+	ginkgo.By("Waiting for daemon set rollout status to finish")
+	statusCheck := []string{"rollout", "status", "daemonset/vsphere-csi-node", "--namespace=" + csiSystemNamespace}
+	framework.RunKubectlOrDie(csiSystemNamespace, statusCheck...)
```

Comment:
we should verify output of this, to check for success, also we should use csi controller deployment here as well

Reply:
code corrected